src/core/fetchers/segment/prioritizer.ts

```diff
@@ -58,16 +48,29 @@ export type ITaskEvent<T> = IInterruptedTaskEvent |
 
 /** Stored object representing a single task. */
 interface IPrioritizerTask<T> {
-  /** Task to run. */
+  /**
+   * Observable created by the ObservablePrioritizer allowing to identify
+   * the task.
+   * This is another Observable wrapping the underlying Observable that the
+   * ObservablePrioritizer is asked to run.
+   */
   observable : Observable<ITaskEvent<T>>;
   /**
-   * Subject used to start and interrupt a task:
-   *   - when `true` is emitted the task should be started
-   *   - when `false` is emitted, the task should be interrupted
+   * Function used to start and interrupt a task:
+   *   - when `true` is emitted the task should be started. If it was already
+   *     started, it will be interrupted first.
+   *   - when `false` is emitted, the task should just be interrupted if running
    */
-  trigger : Subject<boolean>;
+  trigger : (shouldRun : boolean) => void;
```

Comment:
I basically switched from a
```js
trigger = Subject.switchMap(runThatTask)
```
 based approach to a:
```js
function trigger() {
  if (previousOne) {
    cancelPrevious();
  }
  runThatTask();
}
```
based one, because inner Observable are harder to use when creating an Observable from scratch (which was used to add the teardown logic).